src/util/base_type.cpp

```diff
@@ -253,6 +253,34 @@ bool base_type_eqt::base_type_eq_rec(
   }
   else if(type1.id()==ID_pointer)
   {
+    // Types dervied from pointer, such as java_generic_parametert, may have
+    // qualifiers given as named subexpressions:
+    const auto &named_subs1 = type1.get_named_sub();
+    const auto &named_subs2 = type2.get_named_sub();
+
+    for(const auto &name_and_sub : named_subs1)
```

Comment:
Please use two iterators over the two collections, just like `irept::operator==` does. Using `find` is expensive and unnecessary as you can do this with a linear algorithm.